From 6a17924477f64028cc5ff5b51047f3a9fd962a5f Mon Sep 17 00:00:00 2001
From: Jarod Wilson <jarod@redhat.com>
Date: Tue, 7 Feb 2012 11:46:37 -0500
Subject: [PATCH 16/17] [media] lirc_dev: add some __user annotations

Backport from upstream kernel:

commit 0e835087dfe7db19f1f072046f5e116d4ec6662b
Author: Dan Carpenter <error27@gmail.com>
Date:   Wed Nov 17 02:13:39 2010 -0300

    [media] lirc_dev: add some __user annotations

    Sparse complains because there are no __user annotations.

    drivers/media/rc/lirc_dev.c:156:27: warning:
        incorrect type in initializer (incompatible argument 2 (different address spaces))
    drivers/media/rc/lirc_dev.c:156:27:    expected int ( *read )( ... )
    drivers/media/rc/lirc_dev.c:156:27:    got int ( extern [toplevel] *<noident> )( ... )

    Signed-off-by: Dan Carpenter <error27@gmail.com>
    Acked-by: Jarod Wilson <jarod@redhat.com>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>
---
 drivers/lirc_dev/lirc_dev.c | 8 +++-----
 drivers/lirc_dev/lirc_dev.h | 6 +++---
 2 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/drivers/lirc_dev/lirc_dev.c b/drivers/lirc_dev/lirc_dev.c
index 96c9148..5b25c5f 100644
--- a/drivers/lirc_dev/lirc_dev.c
+++ b/drivers/lirc_dev/lirc_dev.c
@@ -748,10 +748,8 @@ long lirc_dev_fop_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 }
 EXPORT_SYMBOL(lirc_dev_fop_ioctl);
 
-ssize_t lirc_dev_fop_read(struct file *file,
-			  char *buffer,
-			  size_t length,
-			  loff_t *ppos)
+ssize_t lirc_dev_fop_read(struct file *file, char __user *buffer,
+			  size_t length, loff_t *ppos)
 {
 	struct irctl *ir = irctls[iminor(file->f_dentry->d_inode)];
 	unsigned char *buf;
@@ -870,7 +868,7 @@ void *lirc_get_pdata(struct file *file)
 EXPORT_SYMBOL(lirc_get_pdata);
 
 
-ssize_t lirc_dev_fop_write(struct file *file, const char *buffer,
+ssize_t lirc_dev_fop_write(struct file *file, const char __user *buffer,
 			   size_t length, loff_t *ppos)
 {
 	struct irctl *ir = irctls[iminor(file->f_dentry->d_inode)];
diff --git a/drivers/lirc_dev/lirc_dev.h b/drivers/lirc_dev/lirc_dev.h
index ca641f8..34056aa 100644
--- a/drivers/lirc_dev/lirc_dev.h
+++ b/drivers/lirc_dev/lirc_dev.h
@@ -257,9 +257,9 @@ int lirc_dev_fop_ioctl(struct inode *inode, struct file *file,
 #else
 long lirc_dev_fop_ioctl(struct file *file, unsigned int cmd, unsigned long arg);
 #endif
-ssize_t lirc_dev_fop_read(struct file *file, char *buffer, size_t length,
+ssize_t lirc_dev_fop_read(struct file *file, char __user *buffer, size_t length,
 			  loff_t *ppos);
-ssize_t lirc_dev_fop_write(struct file *file, const char *buffer, size_t length,
-			   loff_t *ppos);
+ssize_t lirc_dev_fop_write(struct file *file, const char __user *buffer,
+			   size_t length, loff_t *ppos);
 
 #endif
-- 
1.8.3.1

